Doc/library/dataclasses.rst

```diff
@@ -49,26 +49,26 @@ Module contents
 .. decorator:: dataclass(*, init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False, match_args=True, kw_only=False, slots=False, weakref_slot=False)
 
    This function is a :term:`decorator` that is used to add generated
-   :term:`special method`\s to classes, as described below.
+   :term:`special methods <special method>` to classes, as described below.
 
-   The :func:`dataclass` decorator examines the class to find
+   The :func:`!dataclass` decorator examines the class to find
```

Comment:
Although I realise it slightly contradicts @hugovk's advice on Discourse, I'd actually prefer to use ``@dataclass`` for this and similar instances, since the `func` role means it will be rendered as `dataclass()`, which is actually somewhat confusing — it's bad style to do

```py
@dataclass()
class Whatever:
    pass
```

since the parentheses are redundant here

Reply:
I was merely suppressing the link, but I see your point.  There doesn't seem to be a `:decorator:` role, so what is the usual markup?

Reply:
Agreed 👍

Reply:
If `:func:` automatically adds `()`, shouldn't we have `:decorator:` to automatically add `@`?  Perhaps this change can wait for a future update?

Reply:
I'll use this: 
```
:func:`!@dataclass`
```

Reply:
If sphinx is happy with that, I'm happy!

Reply:
(Narrator:) "Sphinx was not happy with it"

I have to figure out how to run the CI warnings check locally.....

Reply:
I _suspected_ it might get grumpy with you about that ;)

Reply:
OK, so this works for non-linked mentions:
```
``@dataclass``
```
(though the original also worked because Sphinx wasn't trying to find "@dataclass".

But what can we do for linked mentions? This is the cause of the nine CI warnings:
```
:func:`@dataclass`
```
So do we use [`dataclass()`](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass) (no @ but easy to make with `:func:`), or do we cobble something ugly together to somehow get the text we want with the link we want?

Reply:
For linked mentions, you can do the (annoyingly verbose -- but it works)

```
:func:`dataclass <@dataclass>`

Reply:
Other way around:
```
:func:`@dataclass <dataclass>`
```
but OK, I've applied it.

Reply:
right, that's what I meant :-)